lib/utils.js

```diff
@@ -29,3 +29,52 @@ exports.shallowObjEqual = function(a, b) {
       a[key] === b[key];
   });
 };
+
+var defaultCompileOptions = {
+  generateLetDeclarations: false,
+  parse: require("./parsers/default.js").parse,
+  ast: false
+};
+
+exports.getOption = function (options, name) {
+  if (options && hasOwn.call(options, name)) {
+    return options[name];
+  }
+  return defaultCompileOptions[name];
+};
+
+exports.getNamesFromPattern = function (pattern) {
+  var queue = [pattern];
+  var names = [];
+
+  for (var i = 0; i < queue.length; ++i) {
+    var pattern = queue[i];
+    if (! pattern) {
+      continue;
+    }
+
+    switch (pattern.type) {
+    case "Identifier":
+      names.push(pattern.name);
+      break;
+    case "Property":
+    case "ObjectProperty":
+      queue.push(pattern.value);
+      break;
+    case "AssignmentPattern":
+      queue.push(pattern.left);
+      break;
+    case "ObjectPattern":
+      queue.push.apply(queue, pattern.properties);
+      break;
+    case "ArrayPattern":
+      queue.push.apply(queue, pattern.elements);
+      break;
+    case "RestElement":
+      queue.push(pattern.argument);
+      break;
+    }
```

Comment:
Because I really didn't want to have to explain how that method works, I took some time this morning to simplify it considerably: https://github.com/benjamn/reify/pull/89